lib/features/details/presentation/view/movie_details_page.dart

```diff
@@ -89,31 +129,46 @@ class _MovieDetailsPageState extends State<MovieDetailsPage> {
                 const SizedBox(height: 12),
                 Align(
                   alignment: Alignment.centerLeft,
-                  child: Container(
-                    height: 45,
-                    width: 120,
-                    decoration: BoxDecoration(
-                        border: Border.all(color: Colors.white),
-                        borderRadius: BorderRadius.circular(12)),
-                    child: const Padding(
-                      padding: EdgeInsets.all(8.0),
-                      child: Row(
-                        children: [
-                          Icon(
-                            Icons.check,
-                            color: Colors.white,
-                          ),
-                          SizedBox(
-                            width: 5,
+                  child: ValueListenableBuilder(
```

Comment:
This ValueListenableBuilder should be directly wrapping the Icon widget where the state change is necessary for the UI to change.
Not this `GestureDetector`.